src/services/endpoint/EndpointService.ts

```diff
@@ -8,31 +8,31 @@ import IQService = angular.IQService;
 import {LaunchTypeService} from "../launchtype/LaunchTypeService";
 import {isNotNullOrUndefined, traverse, deepValue} from "../../common/utils/Utils";
 import {ICacheContainer} from "../container/ICacheContainer";
+import {IServerGroup} from "../server-group/IServerGroup";
+import {CompositeOpBuilder} from "../dmr/CompositeOpBuilder";
+import {Endpoint} from "./Endpoint";
+import {IDmrCompositeReq} from "../dmr/IDmrCompositeReq";
 
 const module: ng.IModule = App.module("managementConsole.services.endpoint", []);
 
 export class EndpointService {
   static $inject: string[] = ["$q", "dmrService", "socketBindingService", "launchType"];
 
-  static parseEndpoint(name: string, object: any, socketBinding?: ISocketBinding): IEndpoint {
-    return <IEndpoint> {
-      name: name,
-      "cache-container": object["cache-container"],
-      encryption: (object.encryption != null && object.encryption !== undefined) ? object.encryption : "",
-      "socket-binding-name": object["socket-binding"],
-      "socket-binding": socketBinding,
-      "hotrod-socket-binding": object["hotrod-socket-binding"],
-      "rest-socket-binding": object["rest-socket-binding"]
-    };
+  static parseEndpoint(namePath: string [], object: any, socketBinding?: ISocketBinding): IEndpoint {
+    return new Endpoint(namePath, object, socketBinding);
   }
 
   constructor(private $q: IQService, private dmrService: DmrService,
               private socketBindingService: SocketBindingService, private launchType: LaunchTypeService) {
   }
 
+  createEndpoint(namePath: string []): IEndpoint {
+    return new Endpoint(namePath, {"is-new-node": true}, undefined);
```

Comment:
This method should be static. Also, if we simply pass undefined for the second arg, then we don't need [this](https://github.com/vblagoje/infinispan-management-console/blob/7f169f8473a89b9fed7067d013a1bf5e3eb3627a/src/components/configuration-section/ConfigurationSectionCtrl.ts#L34) hack. This may need slight changes elsewhere, but the `configuration-section` was designed to handle the creation of new configurations as we do with templates etc.